### ISSUE TITLE
add a function useTree for easy sort or search tree data structure 

### ISSUE THREAD

**Issue by lainlee** [vueuse/vueuse#2718]
### Clear and concise description of the problem

In some case, I need to search or regroup tree data like organization tree, some UI component  can handle this, but most of the time I still handle this myself, so it is necessary to add a function called useTree for handling tree data.

有时候，需要对树形数据进行搜索或者重新分组，比如：组织机构树，有些 UI 框架可以做这些功能，但是更多时候还是得自己处理数据，因此，我认为有必要添加一个针对树形结构处理的方法

### Suggested solution

add a new function called useTree, the params is tree data, return some function like search or sort

创建一个 useTree 方法，传入树形数据，返回一些常用的操作方法

```
const treeData = [
  {
    name: 'main',
    children: [
      { name: '1', children: [...] },
      { name: '2', children: [...] }
    ]
  }
]
const { search } = useTree(treeData)
const result = search({ name: '1' })
```

### Alternative

is there any function can handle these data? or need other return function, plz tell me.

是否已经有合适的方法可以处理这些数据，或者还有其他需要的操作，麻烦大家补充一下

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by Alfred-Skyblue** [comment]
I think this kind of business tends to be customized, it is not recommended to put it in `vueuse`, you can encapsulate the tree structure query method in the project

**Comment by lainlee** [comment]
> I think this kind of business tends to be customized, it is not recommended to put it in `vueuse`, you can encapsulate the tree structure query method in the project

But I think this is just a function working with a specific data structure, like Array, here is the tree data structure, and we have lots of case to use it, such as router structure, or menu, or select.

By the way, is there any rule or standard to determine what function can be added to vueuse?

**Comment by Alfred-Skyblue** [comment]
The business scenario you describe does exist and I am willing to create a PR for it

**Comment by lainlee** [comment]
> The business scenario you describe does exist and I am willing to create a PR for it

Thank you! and I have another question, the PR you created is named useTreeFind which have ability to search tree, but if I want to sort tree, it's better to put this function together or create a new function named useTreeSort?

**Comment by Alfred-Skyblue** [comment]
Do you want to recursively sort the tree structure?

**Comment by lainlee** [comment]
> Do you want to recursively sort the tree structure?

Just an example, but I think there have lots of case with tree structure not only searching, maybe sort or flat or filter.

**Comment by Alfred-Skyblue** [comment]
I think the scene of tree structure sorting is rare. In most cases, this kind of scene needs to be handed over to the server for processing.

**Comment by lainlee** [comment]
> I think the scene of tree structure sorting is rare. In most cases, this kind of scene needs to be handed over to the server for processing.

You are Right, thank you for your coding so quickly and awesome! But still, I want to make tree structure more complete, perhaps I'll create some PR about it in the future.
